live.tsx

```diff
@@ -19,22 +19,66 @@ export const setupLive = (manifest: DecoManifest, liveOptions: LiveOptions) => {
 };
 
 const isDenoDeploy = Deno.env.get("DENO_DEPLOYMENT_ID") !== undefined;
-interface LivePageData {
-  manifest?: any;
-  components?: any;
-  defaultRender?: any;
+
+export interface PageComponentData {
+  component: string;
+  props?: Record<string, unknown>;
+}
+
+export interface LivePageData {
+  components?: PageComponentData[];
 }
 
-export interface LivePageOptions<Data = unknown> {
+export interface LoadLiveComponentsOptions {
   template?: string;
-  render?: (props: PageProps<LivePageData & Data>) => any;
-  loader?: (req: Request, ctx: HandlerContext<Data>) => Promise<Data>;
 }
 
-export function createLivePage<Data>(options: LivePageOptions<Data>) {
-  const defaultRender = options.render;
+export async function loadLiveComponents(
+  req: Request,
+  _: HandlerContext<any>,
+  options?: LoadLiveComponentsOptions,
+): Promise<LivePageData> {
+  const site = userOptions.site;
+  const url = new URL(req.url);
+  const { template } = options ?? {};
+
+  /**
+   * Queries follow PostgREST syntax
+   * https://postgrest.org/en/stable/api.html#horizontal-filtering-rows
+   */
+  const queries = [url.pathname, template]
```

Comment:
will this behave nicely if empty? perform 2 queries?

Reply:
Nope, if template is undefined, only perform 1 query page = url.pathname.